### PR TITLE
Add test for inherited method on functional interface

### PR DESCRIPTION
**Pull request by thk123**
Apologies for my over eager merging of #5013 Added the test @smowton requested in the original PR

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/regression/jbmc/lambda-unhandled-types/test.desc [hunk 1]
```diff
@@ -7,10 +7,13 @@ Test.class
 ^ignoring invokedynamic at java::Test.main:\(\)V address 0 which produces a stub type java::StubInterface
 ^ignoring invokedynamic at java::Test.main:\(\)V address 6 which produces type java::StubSuperinterface which should have exactly one abstract method but actually has 0.
 ^ignoring invokedynamic at java::Test.main:\(\)V address 12 which produces type java::InterfaceDeclaringEquals which should have exactly one abstract method but actually has 2.
-^ignoring invokedynamic at java::Test.main:\(\)V address 18 which produces type java::InterfaceWithDefaults which should have exactly one abstract method but actually has 2.
+^ignoring invokedynamic at java::Test.main:\(\)V address 18 which produces type java::InterfaceWithDefaults
```
Comment (smowton) [inline jbmc/regression/jbmc/lambda-unhandled-types/test.desc]: Accidentally breaking this line?

**Comment by thk123** [comment]
Rebased to fix doxygen problem and address @smowton comment 
@NathanJPhillips @smowton could you both review this thanks?

**Review by smowton — APPROVED** [review]

**Comment by thk123** [comment]
clang format issue addressed (I didn't even know we formatted java files...)

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3a4143e37973186621de8ca6dbe5e3c2e6c41ed0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/124888973

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**